common/src/main/java/us/myles/ViaVersion/protocols/protocol1_13to1_12_2/blockconnections/ConnectionData.java

```diff
@@ -16,195 +16,204 @@
+                            section.setFlatBlock(x, y, z, block);
+                        }
+
+                        if (x == 0) {
+                            update(user, new Position(xOff - 1, yOff + y, zOff + z));
+                        } else if (x == 15) {
+                            update(user, new Position(xOff + 16, yOff + y, zOff + z));
+                        }
+                        if (z == 0) {
+                            update(user, new Position(xOff + x, yOff + y, zOff - 1));
+                        } else if (z == 15) {
+                            update(user, new Position(xOff + x, yOff + y, zOff + 16));
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    public static void init() {
+        if (!Via.getConfig().isServersideBlockConnections()) return;
+        Via.getPlatform().getLogger().info("Loading block connection mappings ...");
+        JsonObject mapping1_13 = MappingData.loadData("mapping-1.13.json");
+        JsonObject blocks1_13 = mapping1_13.getAsJsonObject("blocks");
+        for (Entry<String, JsonElement> blockState : blocks1_13.entrySet()) {
+            Integer id = Integer.parseInt(blockState.getKey());
+            String key = blockState.getValue().getAsString();
+            idToKey.put(id, key);
+            keyToId.put(key, id);
+        }
+
+        if (!Via.getConfig().isReduceBlockStorageMemory()) {
+            JsonObject mappingBlockConnections = MappingData.loadData("blockConnections.json");
+            for (Entry<String, JsonElement> entry : mappingBlockConnections.entrySet()) {
+                int id = keyToId.get(entry.getKey());
+                BlockData blockData = new BlockData();
+                for (Entry<String, JsonElement> type : entry.getValue().getAsJsonObject().entrySet()) {
+                    String name = type.getKey();
+                    JsonObject object = type.getValue().getAsJsonObject();
+                    Boolean[] data = new Boolean[6];
+                    for (BlockFace value : BlockFace.values()) {
+                        String face = value.toString().toLowerCase();
+                        if (object.has(face)) {
+                            data[value.ordinal()] = object.getAsJsonPrimitive(face).getAsBoolean();
+                        } else {
+                            data[value.ordinal()] = false;
+                        }
+                    }
+                    blockData.put(name, data);
+                }
+                blockConnectionData.put(id, blockData);
+            }
+        }
+
+        JsonObject blockData = MappingData.loadData("blockData.json");
+        JsonArray occluding = blockData.getAsJsonArray("occluding");
+        for (JsonElement jsonElement : occluding) {
+            occludingStates.add(keyToId.get(jsonElement.getAsString()));
+        }
+
```

Comment:
Changes (other than formatting) in this file are from this line to 184 and the interface the bottom